make/autoconf/jdk-options.m4

```diff
@@ -438,6 +438,14 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_ADDRESS_SANITIZER],
           # It's harmless to be suppressed in clang as well.
           ASAN_CFLAGS="-fsanitize=address -Wno-stringop-truncation -fno-omit-frame-pointer -fno-common -DADDRESS_SANITIZER"
           ASAN_LDFLAGS="-fsanitize=address"
+          # detect_stack_use_after_return is using a fake stack and thus JDK would crashe:
+          # assert(_thread->is_in_live_stack((address)this)) failed: not on stack?
+          if test "x$TOOLCHAIN_TYPE" = "xgcc"; then
+            ASAN_CFLAGS="$ASAN_CFLAGS --param asan-use-after-return=0"
+          fi
+          if test "x$TOOLCHAIN_TYPE" = "xclang"; then
+            ASAN_CFLAGS="$ASAN_CFLAGS -fsanitize-address-use-after-return=never"
+          fi
```

Comment:
Curious, the other ASAN options work across clang and GCC? Only use-after-return requires different flags for GCC and clang?

Reply:
Most of the options are compatible, for example these (a subset): `-fsanitize=undefined -fno-sanitize-recover=all -fsanitize=float-divide-by-zero -fsanitize=float-cast-overflow -fno-sanitize=null -fno-sanitize=alignment`
But compile-time `use-after-return` setting is not. There are only 8 gcc `--param` `asan-*` keywords, I do not know why.

Reply:
Curious. Thanks for the digging, Jan!